profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -48,36 +63,44 @@ export function UseEffectThrashing({ fetchURL, label }) {
 
 export function UseEffectDerivedCalculation() {
   const [remainder, setReminder] = useState();
-  const [clickedTimes, setClickedTimes] = useState();
+  const [clickedTimes, setClickedTimes] = useState(0);
+  const previousClickedTimes = useRef(clickedTimes);
 
   useEffect(() => {
-    setReminder(clickedTimes % 5);
+    if (previousClickedTimes !== clickedTimes) {
+      setReminder(clickedTimes % 5);
+      previousClickedTimes.current = clickedTimes;
+    }
```

Comment:
It is a bad pattern to do calculations on an effect. It causes extra renders, it should be calculated on each render instead.